src/components/core/GeoChart.tsx

```diff
@@ -81,7 +81,10 @@ export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, IGeoCh
     private geoChartOptions: IGeoChartInnerOptions;
 
     public componentDidUpdate(prevProps: IGeoChartInnerProps) {
-        if (isEqual(prevProps.execution, this.props.execution)) {
+        if (!this.props.execution) {
+            return;
+        }
+        if (prevProps.execution && !this.isNewData(prevProps)) {
```

Comment:
We should move `!this.props.execution` and `prevProps.execution` conditions to `isNewData`
My thought was unclear so please refactor `isNewData` to `shouldGeoChartUpdate` which is more meaningful.